openslides_backend/action/actions/user/user_mixin.py

```diff
@@ -238,3 +238,131 @@ def check_meeting_and_users(
+        for instance in self.instances:
+            meeting_ids: Set[str] = set()
+            for field in ("structure_level_$", "number_$", "vote_weight_$"):
+                if field in instance:
+                    meeting_ids.update(instance[field] or set())
+            all_meetings.append(meeting_ids)
+        if any(all_meetings):
+            self.add_to_history(
+                informations,
+                all_meetings,
+                "User updated in meeting {}",
+                "User updated in multiple meetings",
+            )
+
+        # Group x added/removed
+        all_groups_added: List[Set[int]] = []
+        all_groups_removed: List[Set[int]] = []
+        for instance in self.instances:
+            if "group_$_ids" in instance:
+                group_ids_from_instance = self.get_group_ids_from_instance(instance)
+                group_ids_from_db = self.get_group_ids_from_db(instance)
+                added = group_ids_from_instance - group_ids_from_db
+                removed = group_ids_from_db - group_ids_from_instance
+                all_groups_added.append(added)
+                all_groups_removed.append(removed)
+        check_added = any(all_groups_added)
+        check_removed = any(all_groups_removed)
+        if check_added and check_removed:
+            informations.append("Groups changed")
+        elif check_added:
+            self.add_to_history(
+                informations, all_groups_added, "Group {} added", "Groups changed"
+            )
+        elif check_removed:
+            self.add_to_history(
+                informations, all_groups_removed, "Group {} removed", "Groups changed"
+            )
+
+        # OML/CML changed
+        for instance in self.instances:
+            if (
+                "organization_management_level" in instance
+                or "committee_$_management_level" in instance
+            ):
+                informations.append("OML/CML changed")
+                break
+
+        # Set (in)active
+        for instance in self.instances:
+            if "is_active" in instance:
+                informations.append("Set (in)active")
+                break
+
+        return informations
+
+    def add_to_history(
+        self,
+        informations: List[str],
+        data: List[Set[Any]],
+        single_msg: str,
+        multi_msg: str,
+    ) -> None:
+        # assert data not empty
+        for entries in data:
+            if entries:
+                entry_id = list(entries)[0]
+                break
+
+        if all([set([entry_id]) == entries for entries in data]):
+            informations.append(single_msg.format(entry_id))
+        else:
+            informations.append(multi_msg)
+
+    def get_group_ids_from_db(self, instance: Dict[str, Any]) -> Set[int]:
+        user_fqid = fqid_from_collection_and_id("user", instance["id"])
+        user1 = self.datastore.get(user_fqid, ["group_$_ids"], use_changed_models=False)
```

Comment:
please use more verbose names

Reply:
Use user_prepare_fetch here.
user_prepare_fetch is used to get the meetings from the group_$_ids.
It is part of the prepare step.